enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,893 @@
+---
+title: upgrade-without-registry
+authors:
+- "@jhernand"
+reviewers:
+- "@avishayt"
+- "@danielerez"
+- "@mrunalp"
+- "@nmagnezi"
+- "@oourfali"
+approvers:
+- "@sdodson"
+- "@zaneb"
+api-approvers:
+- "@sdodson"
+- "@zaneb"
+creation-date: 2023-06-29
+last-updated: 2023-07-26
+tracking-link:
+- https://issues.redhat.com/browse/RFE-4482
+see-also:
+- https://issues.redhat.com/browse/OCPBUGS-13219
+- https://github.com/openshift/cluster-network-operator/pull/1803
+replaces: []
+superseded-by: []
+---
+
+# Upgrade without registry
+
+## Summary
+
+Provide an automated mechanism to upgrade a cluster without requiring an image
+registry server.
+
+## Motivation
+
+All these stories are in the context of disconnected clusters with limited
+resources, both in the cluster itself and in the surrounding environment:
+
+- The cluster is not connected to the Internet, or the bandwidth is very limited.
+- It isn't possible to bring up additional machines, even temporarily.
+- The resources of the cluster are limited, in particular in terms of CPU and memory.
```

Comment:
This item from the motivation seems inconsistent with the proposal. Using device monitoring on every node, daemonsets to run pods to manage the images across the cluster, and requiring space on each node for multiple copies of the images (to stage them and then to have them in the crio cache) is all going to be resource intensive.

Reply:
I believe disk capacity is less of a concern, but I agree about the CPU.

Reply:
You are right, this process will intensely use the CPU during the process before the upgrade. But only during that time. Before or after that the only CPU consumer will be the daemon set that monitors the devices (if enabled) and that is quite light.

The main alternative to this approach is to have a registry server running inside the cluster permanently. That will consume probably a similar amount of resources during the upgrade, but needs to be there running for ever.